src/pages/signup-affiliates/components/_signup-status.tsx

```diff
@@ -0,0 +1,348 @@
+                                onSubmit()
+                            }}
+                        >
+                            <Localize translate_text={'_t_Change username_t_'} />
+                        </StyledButton>
+                    </Modal>
+                    <Background />
+                </ProgressModal>
+            )}
+            {signup_status == 'Your website is not a valid entry' && (
+                <ProgressModal>
+                    <Modal is_rtl={is_rtl}>
+                        <Image src={Failed} alt="email" width="100" height="100" />
+                        <Header type="subtitle-1" align="center">
+                            <Localize translate_text={'_t_Signup failed_t_'} />
+                        </Header>
+                        <Header type="paragraph-1" align="center" weight="400">
+                            <Localize translate_text="_t_Your website is not a valid entry. Please enter another:_t_" />
+                        </Header>
+                        <AffiliateInput
+                            type={'text'}
+                            value={affiliate_account.personal_details.website_url}
+                            error={website_url_error}
+                            label={web_site}
+                            placeholder={web_site}
+                            onChange={(e) => {
+                                setWebsiteUrlError(affiliate_validation.website_url(e.target.value))
+                                setAffiliateAccount({
+                                    ...affiliate_account,
+                                    personal_details: {
+                                        ...affiliate_account.personal_details,
+                                        website_url: e.target.value,
+                                    },
+                                })
+                            }}
+                            handleError={() => {
+                                setAffiliateAccount({
+                                    ...affiliate_account,
+                                    personal_details: {
+                                        ...affiliate_account.personal_details,
+                                        website_url: '',
+                                    },
+                                })
+                            }}
+                            required
+                        />
+                        <StyledButton
+                            secondary
+                            onClick={() => {
+                                Analytics?.trackEvent('ce_partner_account_signup_form', {
+                                    action: 'try_submit',
+                                    ...analyticsData,
+                                })
+                                setSignupStatus('')
+                                onSubmit()
+                            }}
+                        >
```

Comment:
Seems like both <AffiliateInput /> and <StyledButton /> can be wrapped and you will not need to write this huge code twice.

Reply:
I decided to make reactivity like in rxjs by next code
`
    const interAction = useCallback(({ action }: Parameters<typeof Analytics.trackEvent>[1]) => {
        Analytics?.trackEvent('ce_partner_account_signup_form', {
            action,
            form_name: 'ce_partner_account_signup_form',
        })
        if (action === 'success_popup_cta' || action === 'close_wizard') {
            window.location.href = 'https://deriv.com/partners/'
        } else if (action === 'try_submit') {
            setSignupStatus('')
            onSubmit()
        } else if (action === 'failed_popup_cta') {
            setSignupStatus('')
        }
    }, [])
    `
 Yes, it's not by singleton, but looks cleanier
 what do you say?

Reply:
I understand the goal, but my major callout was especially about wrapping components.

As far as the above snippet is concerned, you are creating a God-function that performs several types of things for various scenarios. Maybe this can suit your goals this time, but such patterns should be limited.

Reply:
Ohh, I see now. Unfortunately, each modal have big difference compare with others due to the different UI and functionality
by the way god-function removed, I see that it's anti-pattern